### DISCUSSION TITLE
Variables: Global built-in variable for url to the current dashboard (__dashboard.url)

### DISCUSSION THREAD

**Discussion by matthubb** [grafana/grafana#37030] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
`${__dashboard.url}` == location.protocol + '//' + location.host + location.pathname

**Why is this needed**:
To facilitate drill down data links updating template variables on the same dashboard.

For example, it would mean that the datalinks present in https://play.grafana.org/d/27c0IFkGk/data-link-with-dashboard-variable?editPanel=4&orgId=1 wouldn't need to include the uid or slug.

**Comment by matthubb** [comment]
`location.href.split('?')[0]` is probably better to include the port

**Comment by marefr** [comment]
Related to #26381 since `{__dashboard.uid}` is not available and no way to construct the current dashboard URL.

**Comment by dprokop** [comment]
@matthubb you can already reference the same dashboard by using `/d/${__dashboard.uid}?${...}` in your data link. The problem here is that this variable does not pop up in the suggested variables list, hence it's confusing I think. Does that help you? 

Also - what version are you using? I've tested the `/d/${__dashboard.uid}?${...}` behavior on 7.1.5.

**Comment by dprokop** [comment]
@matthubb we have spent some more time with @marefr on this and actually the dashboard.uid variable may not be the optimal solution. I mean it will work, but it will cause a page to reload which is not the best experience. You can use dashboard.uid for now as a workaround, and we will try to fix the issue in upcoming releases.

**Comment by matthubb** [comment]
@dprokop agreed I'd tried just `dashboard.uid` but this results in a reload.
One option might be to let us specify data links starting `?...`, currently this gets converted to `scheme://host/installation?...` dropping out the `/d/uid/slug`.

Ideally (IMO), the data link wouldn't need any dashboard-instance (uid, slug) specific information encoded in it while supporting template-variable drill down without reloads.

In the meanwhile, I'll include the slug and make sure I update it with dashboard renames.

**Comment by matthubb** [comment]
> Also - what version are you using? I've tested the `/d/${__dashboard.uid}?${...}` behavior on 7.1.5.

Sorry, overlooked this question. I'm using 7.1.5. `/d/${__dashboard.uid}?...` does work as a link, but reloads the page.

**Comment by vellala2000** [comment]
when can we expect this issue to be addressesed. i am using 7.3.7 and still see its reloading the page when using /d/${__dashboard.uid}?${...}

**Comment by vdwpsmt** [comment]
Data links are very powerfull, but if you work with  dev/prod  environments or scripted dashboards a built-in variable for the current url would be very convenient.

**Comment by vdwpsmt** [comment]
I don't know what the current situation is, but this syntax seems to solve my problem: 
**/d/${__dashboard.uid}?orgId=${__org}**&from=...
apparently if no url is specified it just uses the current url?
eusing grafana 8.1.1

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). It looks like we have some good solutions here, so I will close this. Feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request the feature.
